src/mstring.cc

```diff
@@ -205,6 +205,24 @@ bool mstring::equals(const mstring &str) const {
     return equals(str.data(), str.length());
 }
 
+bool mstring::requals(const mstring &str) const {
```

Comment:
Eh, that is dead code. I had a quick idea of using another comparer in the cache, because the difference is easier to spot when you compare backwards. However, current comparer (with memcmp inside) is probably good enough.